ZeeHomeScreen/Classes/Components/SectionComposite/SectionCompositeViewController.swift

```diff
@@ -567,13 +567,28 @@ import Zee5CoreSDK
                 if let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath) as? UniversalCollectionViewCell {
                     componentModel.screenConfiguration = screenConfiguration
                     cell.backgroundColor = UIColor.clear
-                    
-                    let _ = cell.setComponentModel(componentModel,
-                    model: componentModel,
-                    view: cell.contentView,
-                    delegate: self,
-                    parentViewController: self)
 
+                    if (cell.componentViewController == nil) {
+                        if let componentViewController = ComponenttFactory.viewController(for: componentModel) {
+                            addChild(componentViewController)
```

Comment:
This was handled previously by factory method, I had to change it to comply to what Apple documentation says. Factory method forced the order of execution as:
-addSubview
-addChildViewController
-didMoveToParentViewController

But the correct order according to Apple should be:
-addChildViewController
-addSubview
-didMoveToParentViewController

Because `addChildViewController` already handles the case with calling `will` method, which might do things on a previously attached view, which in the other case was already altered.

Reply:
SRP fix.